reason-react-native/src/apis/ImageEditor.md

```diff
@@ -1,45 +1,231 @@
+          ~height=windowHeight->pt,
+          ~flexDirection=`column,
+          ~alignItems=`center,
+          ~justifyContent=`center,
+          (),
+        ),
+      "frame":
+        style(
+          ~width=displayWidth->pt,
+          ~height=displayHeight->pt,
+          ~alignItems=`center,
+          ~justifyContent=`center,
+          ~borderWidth=StyleSheet.hairlineWidth,
+          (),
+        ),
+    })
+  );
+
+type state = {
+  path: option(string),
+  imageLoaded: bool,
+};
+
+type action =
+  | SetPath(option(string))
+  | SetImageLoaded;
+
+let imageUri = "https://images.unsplash.com/photo-1520453803296-c39eabe2dab4";
+let uri = ImageEditor.fromUriSource(imageUri);
+
+let handleCropImage = (cropData, send, handler) =>
+  ImageEditor.cropImage(
+    ~uri,
+    ~cropData,
+    ~success=successURI => send(handler(successURI)),
+    ~failure=(~code, ~message) => message->Js.Console.warn,
+  );
+
+[@react.component]
+let make = () => {
+  let (state, dispatch) =
+    React.useReducer(
+      (state, action) =>
+        switch (action) {
+        | SetPath(p) => {...state, path: p}
+        | SetImageLoaded => {...state, imageLoaded: true}
+        },
+      {path: None, imageLoaded: false},
+    );
+
+  let size =
+    ImageEditor.size(
+      ~width=(imageWidth *. 0.5)->floor->truncate,
+      ~height=(imageHeight *. 0.5)->floor->truncate,
+    );
+
+  let cropData = offset =>
+    ImageEditor.cropData(
+      ~offset,
+      ~size,
+      ~resizeMode=`cover,
+      (),
+    );
+
+  <View style=styles##container>
+      <Text>
+        "Click on a quadrant of the image to crop it."->React.string
+      </Text>
+      <View style=styles##frame>
+        <TouchableOpacity
+          onPress={e =>
+            handleCropImage(
+              ImageEditor.offset(
+                ~x=
+                  if (e##nativeEvent##locationX /. displayWidth < 0.5) {
+                    0;
+                  } else {
+                    (0.5 *. imageWidth)->floor->truncate;
+                  },
+                ~y=
+                  if (e##nativeEvent##locationY /. displayHeight < 0.5) {
+                    0;
+                  } else {
+                    (0.5 *. imageHeight)->floor->truncate;
+                  },
+              )
+              ->cropData,
+              dispatch,
+              link =>
+              SetPath(Some(link))
+            )
+          }>
+          <Image
+            source={Image.Source.fromUriSource(
+              Image.uriSource(
+                ~uri=
+                  Belt.Option.getWithDefault(
+                    state.path,
+                    imageUri,
+                  ),
+                ~width=displayWidth,
+                ~height=
+                  (
+                    if (state.imageLoaded) {
+                      1.;
+                    } else {
+                      0.;
+                    }
+                  )
+                  *. displayHeight,
```

Comment:
Why not just put `displayHeight` instead of `1.` above?

Reply:
I was trying different multiples to align the text in the center of the box, but ultimately I had to pick `0.`, so `displayHeight` and `0.` would be cleaner.

Reply:
Removed that artifact